File: backend/moyeo/src/main/java/com/mo/moyeo/domain/currency/entity/Currency.java
package com.mo.moyeo.domain.currency.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity(name = "currencies")
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Data
public class Currency {
    @Id
    @Column(name = "currency_code")
    @Enumerated(EnumType.STRING)
    private CurrencyType id;

    @Column(name = "country_name")
    private String countryName;

    @Column(name="currency_unit")
    private String currencyUnit;

    @Column(name = "country_flag")
    private String countryFlag;
}

File: frontend/app/src/main/java/com/d108/moyeo/domain/model/exchange/Currency.kt
package com.d108.moyeo.domain.model.exchange

import java.math.BigDecimal

data class Currency(
    val currencyCode: String,
    val buyRate: BigDecimal,
    val sellRate: BigDecimal,
    val originalRate: BigDecimal,
    val countryFlag: String?
)
